go/vt/vtgate/planbuilder/operators/operator_funcs.go

```diff
@@ -368,3 +299,104 @@ func BreakExpressionInLHSandRHS(
 	ctx.JoinPredicates[expr] = append(ctx.JoinPredicates[expr], rewrittenExpr)
 	return
 }
+
+type joinOperator interface {
+	Operator
+	getLHS() Operator
+	getRHS() Operator
+	setLHS(Operator)
+	setRHS(Operator)
+	makeInner()
+	isInner() bool
+	addJoinPredicate(ctx *plancontext.PlanningContext, expr sqlparser.Expr) error
+}
```

Comment:
nit: i think we should move this to a better place, like in one of the join files (`apply_join.go` or `join.go`)